### ISSUE TITLE
Exploded-view diagram for Mapbox documentation

### ISSUE THREAD

**Issue by johnnybaloney** [mapbox/mapbox-gl-native#10637]
TL;DR This is a suggestion to enhance Mapbox documentation (e.g. [this page](https://www.mapbox.com/help/how-mapbox-works-overview/)) with an exploded-view diagram (example below) to help new users in particular better understand the layers Mapbox maps consist of and get them going more quickly.

<br/>

It could be due to my limited experience and exposure to Mapbox or I'm just not sharp enough to instantly grok it all but as a newcomer I found it challenging to piece together the components of Mapbox and their equivalents on the API side of things, in this case Android library.

Perhaps a seasoned Mapboxer will not have any difficulty in telling apart a `vector source` from a `source layer` from a `source` but while I can put together a client app that works with Mapbox I'm a bit hazy how it all fits together under the hood.

The documentation is somewhat helpful but there are a number of pages (an example list at the bottom) that one has to go through to create a mental picture of the relations between different pieces. I think that [this page](https://www.mapbox.com/help/define-source-layer/) is a good example of the challenge ahead, as it shows that a `layer` can be a shorthand for a `source layer`, while a `source` is a `tileset source` (at least in this context), and `tileset source` appears to be a composite made up of numerous `source layers`.

In order to guide myself through the sea of all these new terms I sketched this exploded-view diagram which shows at the glance what is going on, as well as giving examples of use on the API side. There are a couple of problems though: first of all I don't know if it's correct, and then I'm still ambiguous whether some terms are right, synonymous with other terms (given in brackets) or separate parts of the jigsaw that I didn't include in the diagram.

```
                     Concept                                     API
                                                 ||
                      style --------------------())--> MapboxMap.setStyleUrl(url)
                        ^                        ||
                        |                        ||    MapboxMap.addLayer(Layer)
                        |                        ||     / Layer:
                        |                        ||       - BackgroundLayer
    (tileset+styling) layer (style layer?)    ~~~||~~~    - LineLayer   (ref: source layer)
                        ^                        ||       - SymbolLayer (ref: source layer)
                        |                        ||       - CircleLayer (ref: source layer)
                        |                        ||       - FillLayer   (ref: source layer)
                        |                        ||       - RasterLayer (ref: source layer)
                        |                        ||       - CustomLayer
                        |                        ||       - UnknownLayer
                        |                        ||
   .--------> tileset source (source layer?)  ~~~||~~~ SomeLayers.setSourceLayer(sourceLayerId)
   |                    ^                        ||
   |                    |                        ||
   |                    |                        ||
   |                    |                        ||
raster      .---> vector tileset (vector source?)||
tileset     |           ^                        ||
   ^        |           |                        ||
   |        |           |                        ||
   |        |           |                        ||
   |        |      vector tiles (source layer?)  ||
 TIFF       |           ^                        ||
            |           |                        ||
            |           |                        ||
            |           |                        ||
            |        dataset (source?)        ~~~||~~~ MapboxMap.addSource(Source)
            |           ^                        ||     / new Layer(id,sourceId)
autoconvert |           |                        ||     / new GeoJsonSource(sourceId)
            |           |                        ||
            |           |                        ||
            |    GeoJSON features                ||
            |           ^                        ||
            |           |                        ||
            |           |                        ||
            |           |                        ||
            '--GPX, GeoJSON, CSV...              ||
```

* A Vector Tile SHOULD contain at least one layer. A layer SHOULD contain at least one feature.
https://github.com/mapbox/vector-tile-spec/tree/master/2.1#41-layers

* A feature refers to either a geospatial feature in a dataset or in a tileset.
https://www.mapbox.com/help/define-features/

* A source layer is a specific layer within a vector source.
https://www.mapbox.com/help/define-source-layer/

* A dataset is an editable collection of GeoJSON features.
https://www.mapbox.com/help/define-dataset/

* A tileset is a collection of raster or vector data. Vector tilesets ... store vector data in the form of points, lines, and polygons as vector tiles.
https://www.mapbox.com/help/define-tileset/

* Vector tiles ... are the key ingredient behind our vector tilesets.
https://www.mapbox.com/help/define-vector-tiles/

* Layers are used in styles to add styling rules to specific subsets of data (=tilesets).
https://www.mapbox.com/help/define-layer/

* "Source" refers to the source object from the style. "Source layer" (as set by setSourceLayer) refers to the Vector Tile layer within the source.
https://github.com/mapbox/mapbox-gl-native/issues/10631#issuecomment-349112364

* A source supplies data to be shown on maps that use Mapbox GL. Sources are defined by the Mapbox Style Specification, and are used with style layers to visually represent the source data on to a map.
https://www.mapbox.com/help/define-source/
